python/dialects/aie.py

```diff
@@ -139,101 +139,82 @@ def __init__(self, tile, link_with=None):
+            consumerTiles=consumerTiles,
+            elemNumber=depth,
+            elemType=of_Ty,
             dimensionsToStream=dimensionsToStream,
             dimensionsFromStreamPerConsumer=dimensionsFromStreamPerConsumer,
         )
 
+    def acquire(self, num_elem):
+        subview_t = ObjectFifoSubviewType.get(self.datatype)
+        acq = ObjectFifoAcquireOp(subview_t, self.sym_name.value, num_elem)
 
-# Create an aie objectFifo acquire op of given number of elements with given memref datatype,
-# from objFifo with given name.
-class ObjectFifoAcquireOp(ObjectFifoAcquireOp):
-    def __init__(self, port, of_name, num_elem, datatype):
-        subview_t = ObjectFifoSubviewType.get(datatype)
-        self.datatype = datatype
-        super().__init__(subview_t, port, of_name, num_elem)
-
-    def acquired_elem(self):
         objects = []
-        if self.size.value == 1:
+        if acq.size.value == 1:
             return ObjectFifoSubviewAccessOp(
-                self.datatype, self.subview, self.size.value - 1
+                self.datatype, acq.subview, acq.size.value - 1
             )
-        for i in range(self.size.value):
-            objects.append(ObjectFifoSubviewAccessOp(self.datatype, self.subview, i))
+        for i in range(acq.size.value):
+            objects.append(ObjectFifoSubviewAccessOp(self.datatype, acq.subview, i))
         return objects
 
-
-def acquire(port, of_name, num_elem, datatype):
-    return ObjectFifoAcquireOp(port, of_name, num_elem, datatype)
+    def release(self, num_elem):
+        return objectfifo_release(self.sym_name.value, num_elem)
 
 
-# Create a flow between source and destination tile ports.
-class Flow(FlowOp):
-    """Specialize FlowOp class constructor to take python integers"""
-
+# Create an aie objectFifo_link between input and output objectFifos.
+class objectfifo_link(ObjectFifoLinkOp):
+    """Specialize ObjectFifoLinkOp class constructor to take python variables"""
     def __init__(
-        self, source, source_port, source_channel, dest, dest_port, dest_channel
+        self,
+        fifoIns,
+        fifoOuts,
     ):
+        if not isinstance(fifoIns, List):
+            fifoIns = [fifoIns]
+        if not isinstance(fifoOuts, List):
+            fifoOuts = [fifoOuts]
+        fifoInRefs = map(
+            lambda i: i if isinstance(i, str) else i.sym_name.value, fifoIns
+        )
+        fifoOutRefs = map(
+            lambda i: i if isinstance(i, str) else i.sym_name.value, fifoOuts
+        )
         super().__init__(
-            source=source,
-            sourceBundle=source_port,
-            sourceChannel=source_channel,
-            dest=dest,
-            destBundle=dest_port,
-            destChannel=dest_channel,
+            fifoIns=fifoInRefs,
+            fifoOuts=fifoOutRefs,
         )
 
 
 # Create a packet flow between source and destination tile ports.
-class PacketFlow(PacketFlowOp):
+class packet_flow(PacketFlowOp):
```

Comment:
This is not a good idea - lowercased names (with no `Op`) are reserved for "value builders" https://github.com/llvm/llvm-project/pull/68308 so this class will shadow an already generated function (you can grep in `_aie_ops_gen.py`).

Reply:
I see, thank you Maks! I will adjust the name to something else that isn't the same as the value builders. I would still like to keep it lowercase since every other function call in the .py examples is lowercase.